lib/faktory/client.rb

```diff
@@ -160,12 +160,20 @@ def fail(jid, ex)
     # Sends a heartbeat to the server, in order to prove this
     # worker process is still alive.
     #
+    # You can pass in the current_state of the process, for example during shutdown
+    # quiet and/or terminate can be supplied. 
+    #
     # Return a string signal to process, legal values are "quiet" or "terminate".
     # The quiet signal is informative: the server won't allow this process to FETCH
     # any more jobs anyways.
-    def beat
+    def beat(current_state = nil)
       transaction do
-        command("BEAT", %Q[{"wid":"#{@@random_process_wid}"}])
+        if current_state.present?
```

Comment:
@mperham let me know if you can think of a cleaner way of doing this, as it seems rather messy

Reply:
Use `.nil?`, `.present?` is Rails-specific. Yeah, I don't see an easy way to clean this up so ugly but explicit is better.